### PR TITLE
Feature: added fst serialization

### PR DESCRIPTION
**Pull request by a364176773**
### Ⅰ. Describe what this PR did
added fst serialization

### Ⅳ. Describe how to verify it
transport.serialization=fst

### CODE REVIEW

File: serializer/seata-serializer-fst/src/test/java/io/seata/serializer/fst/KryoSerializerTest.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+import io.seata.core.protocol.transaction.BranchCommitResponse;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * @author funkye
+ */
+public class KryoSerializerTest {
```
Comment (slievrly) [inline serializer/seata-serializer-fst/src/test/java/io/seata/serializer/fst/KryoSerializerTest.java]: Kryo？

Comment (a364176773) [reply serializer/seata-serializer-fst/src/test/java/io/seata/serializer/fst/KryoSerializerTest.java]: > Kryo？

already edited

File: serializer/seata-serializer-fst/pom.xml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    <dependencies>
+        <dependency>
+            <groupId>de.ruedigermoeller</groupId>
+            <artifactId>fst</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.seata</groupId>
+            <artifactId>seata-core</artifactId>
+            <version>${project.version}</version>
+            <scope>compile</scope>
```
Comment (slievrly) [inline serializer/seata-serializer-fst/pom.xml]: default is compile？

File: core/src/main/java/io/seata/core/serializer/SerializerType.java [hunk 3]
```diff
@@ -42,6 +42,13 @@
      * Math.pow(2, 2)
      */
     KRYO((byte)0x4),
+
+    /**
+     * The fst.
+     * <p>
+     * Math.pow(2, 3)
+     */
+    FST((byte)0x6),
```
Comment (jsbxyyx) [inline core/src/main/java/io/seata/core/serializer/SerializerType.java]: Math.pow(2, 3) = 8

Comment (a364176773) [reply core/src/main/java/io/seata/core/serializer/SerializerType.java]: > Math.pow(2, 3) = 8

already edited PTAL

File: serializer/seata-serializer-fst/pom.xml (hunk 2 above)
Comment (a364176773) [reply serializer/seata-serializer-fst/pom.xml]: > default is compile？

have deleted

File: serializer/seata-serializer-fst/src/main/java/io.seata.serializer.fst/FstSerializer.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import io.seata.common.loader.LoadLevel;
+import io.seata.core.serializer.Serializer;
+
+/**
+ * @author funkye
+ */
+@LoadLevel(name = "FST")
+public class FstSerializer implements Serializer {
+
+    private final FSTConfiguration conf = FSTConfiguration.createDefaultConfiguration();
```
Comment (jsbxyyx) [inline serializer/seata-serializer-fst/src/main/java/io.seata.serializer.fst/FstSerializer.java]: static ? need register class?

**Pull request merged**